src/analyzer/analyzersilence.h

```diff
@@ -0,0 +1,33 @@
+#ifndef ANALYZER_ANALYZERSILENCE_H
+#define ANALYZER_ANALYZERSILENCE_H
+
+#include "analyzer/analyzer.h"
+#include "preferences/usersettings.h"
+
+class CuePointer;
+
+class AnalyzerSilence : public Analyzer {
+  public:
+    explicit AnalyzerSilence(UserSettingsPointer pConfig);
+    ~AnalyzerSilence() override = default;
+
+    bool initialize(TrackPointer tio, int sampleRate, int totalSamples) override;
+    bool isDisabledOrLoadStoredSuccess(TrackPointer tio) const override;
+    void process(const CSAMPLE* pIn, const int iLen) override;
+    void finalize(TrackPointer tio) override;
+    void cleanup(TrackPointer tio) override;
+
+  private:
+    UserSettingsPointer m_pConfig;
+    float m_fThreshold;
+    int m_iFramesProcessed;
+    bool m_bPrevSilence;
+    int m_iSignalStart;
+    int m_iSignalEnd;
+
+    TrackPointer m_pTrack;
```

Comment:
You don't need to store a TrackPointer and you should not modify the track while processing. If the analysis is aborted you end up with an inconsistent state!

The track must be modified in finalize() which provides the TrackPointer.